lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts

```diff
@@ -0,0 +1,54 @@
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import { readYamlDocument } from "@aws-quickstart/eks-blueprints/dist/utils";
+import { Construct } from 'constructs';
+import { CfnRuleGroupsNamespace } from "aws-cdk-lib/aws-aps";
+import { assert } from 'console';
+
+/**
+ * Configuration options for add-on
+ */
+export interface AmpRulesConfiguratorAddOnProps {
+    /** 
+     * AMP workspace ARN.
+     */
+    ampWorkspaceArn: string;
+
+    /** 
+     * Paths of the files listing the AMP rules.
+     */
+    ruleFilePaths: string[];
+}
+
+/**
+ * Implementation of Rules configuration for AMP
+ */
+export class AmpRulesConfiguratorAddOn implements blueprints.ClusterAddOn {
```

Comment:
If it is an extensions addon, then docs are needed.  Also, don't you require a dependency on the AmpAddOn here? Please confirm.

Reply:
@shapirov103 @elamaran11 thanks for your comments, this one is common from both of you. Answering here:
actually the only dependency for this addon is the workspace in AMP, which is provided by:
'''
.resourceProvider(ampWorkspaceName, new blueprints.CreateAmpProvider(ampWorkspaceName, ampWorkspaceName))
'''

the AMP addon deploys the OpenTelemetryCollector for AMP exporting and related permissions. (Maybe a better name would be OpenTelemetryCollectorForAMPAddOn?) so I believe there is no dependency: you can AmpRulesConfiguratorAddOn without AMP addon and it still works because it only interacts with AMP, not the OpenTelemetryCollector. Anything I am missing?